packages/mgt-components/src/components/mgt-file-list/mgt-file-list.ts

```diff
@@ -0,0 +1,592 @@
+        files = await getFilesByListQuery(graph, this.fileListQuery);
+      } else if (this.fileQueries && this.fileQueries !== null) {
+        files = await getFilesByQueries(graph, this.fileQueries);
+      } else if (getFromMyDrive) {
+        if (this.itemId && this.itemId !== null) {
+          files = await getFilesById(graph, this.itemId);
+        } else if (this.itemPath && this.itemPath !== null) {
+          files = await getFilesByPath(graph, this.itemPath);
+        } else if (this.insightType && this.insightType !== null) {
+          files = await getMyInsightsFiles(graph, this.insightType);
+        } else {
+          files = await getFiles(graph);
+        }
+      } else if (this.driveId && this.driveId !== null) {
+        if (this.itemId && this.itemId !== null) {
+          files = await getDriveFilesById(graph, this.driveId, this.itemId);
+        } else if (this.itemPath && this.itemPath !== null) {
+          files = await getDriveFilesByPath(graph, this.driveId, this.itemPath);
+        }
+      } else if (this.groupId && this.groupId !== null) {
+        if (this.itemId && this.itemId !== null) {
+          files = await getGroupFilesById(graph, this.groupId, this.itemId);
+        } else if (this.itemPath && this.itemPath !== null) {
+          files = await getGroupFilesByPath(graph, this.groupId, this.itemPath);
+        }
+      } else if (this.siteId && this.siteId !== null) {
+        if (this.itemId && this.itemId !== null) {
+          files = await getSiteFilesById(graph, this.siteId, this.itemId);
+        } else if (this.itemPath && this.itemPath !== null) {
+          files = await getSiteFilesByPath(graph, this.siteId, this.itemPath);
+        }
+      } else if (this.userId && this.userId !== null) {
+        if (this.itemId && this.itemId !== null) {
+          files = await getUserFilesById(graph, this.userId, this.itemId);
+        } else if (this.itemPath && this.itemPath !== null) {
+          files = await getUserFilesByPath(graph, this.userId, this.itemPath);
+        } else if (this.insightType && this.insightType !== null) {
+          files = await getUserInsightsFiles(graph, this.userId, this.insightType);
+        }
+      }
+
+      this.files = files;
+
+      // filter files when extensions are provided
+      let filteredByFileExtension: MicrosoftGraph.DriveItem[];
+      if (this.fileExtensions && this.fileExtensions !== null) {
+        filteredByFileExtension = files.filter(file => {
+          for (const e of this.fileExtensions) {
+            if (e == this.getFileExtension(file.name)) {
+              return file;
+            }
+          }
+        });
+      }
+
+      if (filteredByFileExtension && filteredByFileExtension.length >= 0) {
+        this.files = filteredByFileExtension;
+      }
+
+      this._renderedFileCount = this.showMax;
+      this._filesToRender = this.files.slice(0, this.showMax);
+    }
+  }
+
+  private showMoreFiles(count: number) {
```

Comment:
Update this function to call the graph for additional  items - currently all files are fetched at once and this can cause performance issues for a large list of files